pkg/cli/credential/aws_credential_management.go

```diff
@@ -77,28 +99,57 @@ func (cpm *AWSCredentialManagementClient) Put(ctx context.Context, credential uc
 // "Get" retrieves the credentials for the specified cloud provider from the backend and returns a ProviderCredentialConfiguration
 // object containing the credentials or an error if the credentials could not be retrieved.
 func (cpm *AWSCredentialManagementClient) Get(ctx context.Context, credentialName string) (ProviderCredentialConfiguration, error) {
-	var err error
-
 	// We send only the name when getting credentials from backend which we already have access to
 	resp, err := cpm.AWSCredentialClient.Get(ctx, AWSPlaneName, credentialName, nil)
 	if err != nil {
 		return ProviderCredentialConfiguration{}, err
 	}
-	awsAccessKeyCredentials, ok := resp.AwsCredentialResource.Properties.(*ucp.AwsAccessKeyCredentialProperties)
-	if !ok {
+
+	switch *resp.AwsCredentialResource.Properties.GetAwsCredentialProperties().Kind {
+	case ucp.AWSCredentialKindAccessKey:
+		awsAccessKeyCredentials, ok := resp.AwsCredentialResource.Properties.(*ucp.AwsAccessKeyCredentialProperties)
+		if !ok {
+			return ProviderCredentialConfiguration{}, clierrors.Message("Unable to find credentials for cloud provider %s.", AWSCredential)
+		}
+
+		providerCredentialConfiguration := ProviderCredentialConfiguration{
+			CloudProviderStatus: CloudProviderStatus{
+				Name:    AWSCredential,
+				Enabled: true,
+			},
+			AWSCredentials: &AWSCredentialProperties{
+				Kind: (*string)(awsAccessKeyCredentials.Kind),
+				AccessKey: &AWSAccessKeyCredentialProperties{
+					AccessKeyID: awsAccessKeyCredentials.AccessKeyID,
+				},
+			},
+		}
+		return providerCredentialConfiguration, nil
+	case ucp.AWSCredentialKindIRSA:
+		awsIRSACredentials, ok := resp.AwsCredentialResource.Properties.(*ucp.AwsIRSACredentialProperties)
+		if !ok {
+			return ProviderCredentialConfiguration{}, clierrors.Message("Unable to find credentials for cloud provider %s.", AWSCredential)
+		}
+
+		providerCredentialConfiguration := ProviderCredentialConfiguration{
+			CloudProviderStatus: CloudProviderStatus{
+				Name:    AWSCredential,
+				Enabled: true,
+			},
+			AWSCredentials: &AWSCredentialProperties{
+				Kind: (*string)(awsIRSACredentials.Kind),
```

Comment:
Same question here as above.

Reply:
we cannot store a credential with out kind, since the data model would not permit it and throw an error saying invalid kind. Therefore upon get, this field cannnot be null.

"code": "HttpRequestPayloadAPISpecValidationFailed",
        "message": "HTTP request payload failed validation against API specification with one or more errors. Please see details for more information.",
        "target": "ucp/openapi",
        "details": [
            {
                "code": "InvalidProperties",
                "message": "$.properties.kind in body should be one of [AccessKey IRSA]"
            }
        ]